classes/Link.php

```diff
@@ -442,7 +442,10 @@ public function getCategoryLink(
             $params['id'] = $category->id;
         } elseif (isset($category['id_category'])) {
             $params['id'] = $category['id_category'];
-        } elseif (is_int($category) or ctype_digit($category)) {
+        } elseif (
+            is_int($category)
+            || (!is_array($category) && !is_a($category, CategoryCore::class) && ctype_digit($category))
```

Comment:
```suggestion
            || is_string($category) && ctype_digit($category))
```

Reply:
(I check it)

Reply:
That doesn't work because the PHPDoc is `CategoryCore|array|int` for `$category`

Reply:
Then you can't use `ctype_digit` as starting with php 8.1 it only allows `string` as argument

Reply:
Oh !!!!!!!!! It's fixed.